recipes/server_check.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require_relative './server_check_helper'
+
+consul_service_file = '/etc/systemd/system/consul.service'
+consul_service_folder = '/etc/systemd/system/consul.service.d'
+drop_in_file = "#{consul_service_folder}/server_start_prevention.conf" # prevent systemd from starting
+metric = %(rogue_consul_server{dc="#{node['consul']['config']['datacenter']}",ip="#{node['ipaddress']}"} 1)
+
+directory consul_service_folder do
+  owner 'root'
+  mode '0644'
+  action :create
+end
+
+# Output file is used for systemd to check whether it should start consul.
+if is_server_allowed?(node['consul']['config']['retry_join'], node['ipaddress'])
+  # For non rogue servers, simply remove the drop-in file to unblock service from starting.
+  if File.exist?(drop_in_file)
+    file drop_in_file do
+      action :delete
+    end
+
+    reload_systemd()
+  end
+else
+  # Stop the service if it is already running.
+  service 'consul' do
+    action :stop
+    only_if { ::File.exist?(consul_service_file) }
+  end
+
+  # Update config itself to further prevent it from becoming a server.
+  node.override['consul']['config']['server'] = false
+
+  # Currently has to use ConditionPathExists instead of ExecCondition.
+  # Blocks the service from starting if this file itself is present.
+  # ExecCondition requires systemd 243+, Ubuntu 1804 has systemd 237.
+  file drop_in_file do
+    owner 'root'
+    mode '0644'
+    content <<-EOF
+        [Unit]
+        ConditionPathExists=!#{drop_in_file}
+        EOF
+    action :create
+  end
+
+  # Send a binary value as metric to indicate the server did not start.
+  unless node['consul']['metrics_write_endpoint'].nil?
+    http_request 'send metrics' do
```

Comment:
I wonder if writing something to the grafana-agent's textfiles is better (or something to do in addition).

This POST can fail (after 5 retries).

But textfiles are forever! I'm also unclear what this metrics call does.

grafana-agent will constantly report there's a misconfigured node which works much better with how prom/vm alerting works.

Reply:
Thanks Sherman, let me look up grafana-agent and how to use that.